index.js

```diff
@@ -26,6 +26,68 @@ module.exports = function (fastify, opts, next) {
     return err
   }
 
+  function loadDirs (dirs, cb, plugins, prefix) {
+    if (dirs.length === 0) {
+      cb(null, plugins)
+      return
+    }
+
+    const toLoad = dirs.pop()
+    fs.stat(toLoad, (err, stat) => {
+      if (err) {
+        cb(err)
+        return
+      }
+
+      if (stat.isDirectory()) {
+        fs.readdir(toLoad, (err, files) => {
+          if (err) {
+            cb(err)
+            return
+          }
+
+          prefix = path.join(prefix, toLoad.split(path.sep).pop())
+          const fileList = files.join('\n')
+          // if the directory does not contain a package.json or an index,
+          // load each script file as an independend plugin
+          if (
+            !packagePattern.test(fileList) &&
+            !indexPattern.test(fileList) &&
+            scriptPattern.test(fileList)
+          ) {
+            for (let index = 0; index < files.length; index++) {
+              const file = files[index]
+
+              if (scriptPattern.test(file)) {
+                plugins.push({
+                  skip: false,
+                  opts: { prefix },
+                  file: path.join(toLoad, file)
+                })
+              } else {
+                dirs.push(path.join(toLoad, file))
+              }
+            }
+          } else {
+            plugins.push({
+              // skip directories without script files inside
+              skip: files.every(name => !scriptPattern.test(name)),
+              file: toLoad
+            })
+          }
+          loadDirs(dirs, cb, plugins, prefix)
```

Comment:
Just to share: I'm asking myself if this should be an opt-in option or just "load all the folders".

Right now I think it is fine 👍

Reply:
reading the issue comments my idea was:
add an option called `recursive` with boolean value, if `true` it loads recursively all dirs

to customize the folder loading the user can add an `autoload.json` file with some options.

Is that correct?

Reply:
I like the 'recursive' flag

I would avoid the autoload.json config.
I think, for example an option sync function filter(path) {} where the user return true or false if he wants to load that folder for example

But I would go through steps 👍